src/tensor/mod.rs

```diff
@@ -575,15 +576,25 @@ macro_rules! try_from_impl {
                     None => failure::bail!("cannot convert to slice"),
                     Some(v) => v,
                 };
-                let tn = Self::of_slice(slice);
+                let tn = Self::f_of_slice(slice)?;
                 let shape: Vec<i64> = value.shape().iter().map(|s| *s as i64).collect();
                 Ok(tn.reshape(&shape))
```

Comment:
I cannot give it a try right now but maybe it would make sense to even use `f_reshape` here?

Reply:
Sounds good. Change it accordingly.